test/jdk/java/awt/print/PrinterJob/PrintNullString.java

```diff
@@ -21,308 +21,203 @@
+                    + " Test cannot continue.");
+        }
+
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .testUI(PrintNullString::new)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build()
+                .awaitAndCheck();
+    }
 
-    c = new TextCanvas();
-    add("Center", c);
+    public PrintNullString() {
+        super("JDK 1.2 drawString Printing");
 
-    Button printButton = new Button("Print");
-    printButton.addActionListener(this);
-    add("South", printButton);
+        c = new TextCanvas();
+        add("Center", c);
 
-    addWindowListener(new WindowAdapter() {
-       public void windowClosing(WindowEvent e) {
-             System.exit(0);
-            }
-    });
+        Button printButton = new Button("Print");
+        printButton.addActionListener(this);
+        add("South", printButton);
 
-    pack();
- }
+        pack();
+    }
 
- public void actionPerformed(ActionEvent e) {
+    public void actionPerformed(ActionEvent e) {
 
-   PrinterJob pj = PrinterJob.getPrinterJob();
+        PrinterJob pj = PrinterJob.getPrinterJob();
 
-   if (pj != null && pj.printDialog()) {
+        if (pj != null && pj.printDialog()) {
 
-       pj.setPrintable(c);
-       try {
-            pj.print();
-      } catch (PrinterException pe) {
-      } finally {
-         System.err.println("PRINT RETURNED");
-      }
-   }
- }
+            pj.setPrintable(c);
+            try {
+                pj.print();
+            } catch (PrinterException pe) {
+            } finally {
+                System.out.println("PRINT RETURNED");
+            }
+        }
+    }
 
- class TextCanvas extends Panel implements Printable {
+    static class TextCanvas extends Panel implements Printable {
 
-    String nullStr = null;
-    String emptyStr = new String();
-    AttributedString nullAttStr = null;
-    AttributedString emptyAttStr = new AttributedString(emptyStr);
-    AttributedCharacterIterator nullIterator = null;
-    AttributedCharacterIterator emptyIterator = emptyAttStr.getIterator();
+        String nullStr = null;
+        String emptyStr = "";
+        AttributedString emptyAttStr = new AttributedString(emptyStr);
+        AttributedCharacterIterator nullIterator = null;
+        AttributedCharacterIterator emptyIterator = emptyAttStr.getIterator();
 
-    public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
+        public int print(Graphics g, PageFormat pgFmt, int pgIndex) {
 
-      if (pgIndex > 0)
-         return Printable.NO_SUCH_PAGE;
+            if (pgIndex > 0)
+                return Printable.NO_SUCH_PAGE;
 
-      Graphics2D g2d = (Graphics2D)g;
-      g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
+            Graphics2D g2d = (Graphics2D) g;
+            g2d.translate(pgFmt.getImageableX(), pgFmt.getImageableY());
```

Comment:
dispose g2D object

Reply:
Do not dispose of the `Graphics` object that is passed as a parameter to `paint` or `print`.